### PR TITLE
Ele 409 add alerts to e2e

### PR DESCRIPTION
**Pull request by RoiTabach**
Classes Added:
* Mocks for slack web client that only stores inside of it the alerts and files "sent"
* Mock for AlertsFetcher that reads normally but doesn't update as sent / skipped in the db, instead stores inside.
* Mock for DataMonitoringAlerts that uses above classes

Functions Added:
* simple test run that verifies the run results are as expected (compare to what alerts that the E2E package gives).

### CODE REVIEW

File: tests/e2e/monitor/test_slack_alerts.py [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+            f"TABLES_TO_UPDATE_IN_EDR_RUN_ON_E2E['{warehouse_type}']={list(data_monitoring.alerts_api.sent_alerts.keys())}"
+        )
+        print(
+            f"EXPECTED_MESSAGE_HEADERS_COUNT['{warehouse_type}']={sorted(Counter([try_parse_header_text_from_slack_message_schema(x) for x in data_monitoring.slack_client.sent_messages[SLACK_CHANNEL_TEST_NOT_USED_NAME]]).items())}"
+        )
+        # this flow shouldn't be left alone in production so assert False to get some attention
+        # not writing assert False  to help whoever reads the error message.
+        assert warehouse_type in NUM_ALERTS_E2E_GIVES.keys()
+
+    if data_monitoring.sent_alert_count != NUM_ALERTS_E2E_GIVES[warehouse_type]:
```
Comment (IDoneShaveIt) [inline tests/e2e/monitor/test_slack_alerts.py:130]: Don't we need an assertion for that as well?

**Review by IDoneShaveIt — COMMENTED** [review]
Looking good overall.
Left some comments

File: tests/mocks/slack_client_mock.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    def __init__(
+        self, token: str = None, webhook: str = None, tracking: AnonymousTracking = None
+    ) -> None:
+        self.sent_messages: List[
+            SlackMessageSchema
+        ] = []  # List of alerts that were sent.
+        super().__init__(token=token, webhook=webhook, tracking=None)
+
+    @sleep_and_retry
+    @limits(calls=50, period=ONE_MINUTE)
```
[the remark this answers is not in the document]
Comment (RoiTabach) [reply tests/mocks/slack_client_mock.py:60]: Good Q. Removing.

File: tests/mocks/slack_client_mock.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+
+class SlackWebhookClientMock(SlackWebhookClient):
+    def __init__(
+        self, token: str = None, webhook: str = None, tracking: AnonymousTracking = None
+    ) -> None:
+        self.sent_messages: List[
+            SlackMessageSchema
+        ] = []  # List of alerts that were sent.
+        super().__init__(token=token, webhook=webhook, tracking=None)
```
[the remark this answers is not in the document]
Comment (RoiTabach) [reply tests/mocks/slack_client_mock.py:57]: My original thought was:

I want to cover as much as I can from e2e-ness without the part that checks if messages were sent to slack. By keeping the real token+ actually initializing the slack client with it- there's a chance to catch something going bad in those parts. 

WDYT?

[the remark this answers is not in the document]
Comment (IDoneShaveIt) [reply tests/mocks/slack_client_mock.py:57]: Sure sounds right.

**Pull request closed without merging**